tests/init/neg/closureLeak.check

```diff
@@ -1,7 +1,8 @@
 -- Error: tests/init/neg/closureLeak.scala:11:14 -----------------------------------------------------------------------
 11 |    l.foreach(a => a.addX(this)) // error
    |              ^^^^^^^^^^^^^^^^^
-   |             Cannot prove that the value is fully initialized. Only initialized values may be used as arguments.
-   |
-   |             The unsafe promotion may cause the following problem:
-   |             Cannot prove that the value is fully initialized. Only initialized values may be used as arguments.
+   |              Cannot prove the argument is fully initialized.
```

Comment:
I still find these messages difficult to understand. Here's how I would explain this particular failure to someone. (We can think about making the error messages closer to such an explanation.)

1. The arguments of a method call in a constructor must be fully initialized. I could not prove that the underlined argument is fully initialized.
2. For a closure to be considered fully initialized, its body must access only fully initialized variables. I could not prove that the underlined access is fully initialized.

One general distinction I see is between "X must be fully initialized because that is the rule (e.g. argument to a method call in a constructor)" vs. "X must be fully initialized in order to prove that Y is fully initialized (e.g. expression in body of closure)".